libwayshot/src/lib.rs

```diff
@@ -142,6 +150,147 @@ impl WayshotConnection {
         state.outputs
     }
 
+    /// Get a FrameCopy instance with screenshot pixel data for any wl_output object.
+    pub fn capture_output_frame_shm_fd<T: AsFd>(
+        &self,
+        cursor_overlay: i32,
+        output: WlOutput,
+        transform: Transform,
+        fd: T,
+        capture_region: Option<CaptureRegion>,
+    ) -> Result<FrameCopy> {
+        // Connecting to wayland environment.
+        let mut state = CaptureFrameState {
+            formats: Vec::new(),
+            state: None,
+            buffer_done: AtomicBool::new(false),
+        };
+        let mut event_queue = self.conn.new_event_queue::<CaptureFrameState>();
+        let qh = event_queue.handle();
+
+        // Instantiating screencopy manager.
+        let screencopy_manager = match self.globals.bind::<ZwlrScreencopyManagerV1, _, _>(
+            &qh,
+            3..=3,
+            (),
+        ) {
+            Ok(x) => x,
+            Err(e) => {
+                log::error!("Failed to create screencopy manager. Does your compositor implement ZwlrScreencopy?");
+                panic!("{:#?}", e);
```

Comment:
e.. I think here should not panic.. if is used in xdg-desktop-portal

Reply:
That is a good point, we can return an error here?